hivemind/dht/__init__.py

```diff
@@ -62,7 +62,7 @@ def __init__(
         *,
         start: bool,
         daemon: bool = True,
-        max_workers: Optional[int] = None,
+        max_workers: int = DHT_DEFAULT_NUM_WORKERS,
```

Comment:
```suggestion
        max_workers: Optional[int] = None,
```

I'd suggest to avoid duplicating the default value here. That is, let's pass `max_workers: Optional[int]` to `DHTNode` directly and set the default value from the env variable if `num_workers is None`.